lighthouse-core/gather/gatherers/obscuring-elements.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const pageFunctions = require('../../lib/page-functions.js');
+const TraceProcessor = require('../../lib/tracehouse/trace-processor.js');
+const { getOuterHTMLSnippet } = require('../../lib/page-functions.js');
+
+/** @typedef {{element?: string, x?: number, y?:number, retList?: Array<any>}} MyData */
+/** @typedef {{xCoord: number, yCoord: number}} Point */
+
+/**
+ * @this {HTMLElement}
+ * @return {LH.Artifacts['ObscuringElements']}
+ */
+/* istanbul ignore next */
+function checkIntersection() {
+  const lcpElement = this.nodeType === document.ELEMENT_NODE ? this : this.parentElement; // eslint-disable-line no-undef
+  const lcpHTMLSnippet = lcpElement && getOuterHTMLSnippet(lcpElement);
+  const obscuringElements = [];
+  const seen = new Set();
```

Comment:
Cool, didn't know about WeakSets. Also, see previous comment as to why I was using snippets.

Reply:
+1 to using elements in the set instead of snippets.

I don't think WeakSet is relevant here. `seen` is local to this function call and is eligible for cleanup at the end of the function. We aren't holding on to it, so there is no memory leak concern.